src/lib/functions/rgb_hex.ts

```diff
@@ -0,0 +1,19 @@
+const to_hex = (
+	red: number,
+	green: number,
+	blue: number,
+	alpha: string
+) => "#" + ((blue | green << 8 | red << 16) | 1 << 24).toString(16).slice(1) + alpha;
+
+export function rgb_hex(rgb: number[], alpha: number) {
+	if (alpha >= 0 && alpha <= 1) {
+		alpha = Math.round(255 * alpha);
+	} else if (alpha >= 0 && alpha <= 100) {
+		alpha = Math.round(255 * alpha / 100);
+	} else {
+		throw new TypeError(`Expected alpha value (${alpha}) as a fraction or number`);
+	};
+
+	const new_alpha = (alpha | 1 << 8).toString(16).slice(1);
+	return to_hex(rgb[0], rgb[1], rgb[2], new_alpha);
+};
```

Comment:
This is where i wanted to use [`rgb-hex`](https://www.npmjs.com/package/rgb-hex)

As it supports percent `rgbHex(40, 42, 54, '75%');`

```ts
// TODO: Remove this ignore comment.

// eslint-disable-next-line no-mixed-operators

const toHex = (red, green, blue, alpha) => ((blue | green << 8 | red << 16) | 1 << 24).toString(16).slice(1) + alpha;


export default function rgbHex(red, green, blue, alpha) {

	let isPercent = (red + (alpha || '')).toString().includes('%');


	if (typeof red === 'string' && !green) { // Single string parameter.

		const parsed = parseCssRgbString(red);

		if (!parsed) {

			throw new TypeError('Invalid or unsupported color format.');

		}


		isPercent = false;

		[red, green, blue, alpha] = parsed;

	} else if (alpha !== undefined) {

		alpha = Number.parseFloat(alpha);

	}


	if (typeof red !== 'number' ||

		typeof green !== 'number' ||

		typeof blue !== 'number' ||

		red > 255 ||

		green > 255 ||

		blue > 255

	) {

		throw new TypeError('Expected three numbers below 256');

	}


	if (typeof alpha === 'number') {

		if (!isPercent && alpha >= 0 && alpha <= 1) {

			alpha = Math.round(255 * alpha);

		} else if (isPercent && alpha >= 0 && alpha <= 100) {

			alpha = Math.round(255 * alpha / 100);

		} else {

			throw new TypeError(`Expected alpha value (${alpha}) as a fraction or percentage`);

		}


		alpha = (alpha | 1 << 8).toString(16).slice(1); // eslint-disable-line no-mixed-operators

	} else {

		alpha = '';

	}


	return toHex(red, green, blue, alpha);

}


const parseCssRgbString = input => {

	const parts = input.replace(/rgba?\(([^)]+)\)/, '$1').split(/[,\s/]+/).filter(Boolean);

	if (parts.length < 3) {

		return;

	}


	const parseValue = (value, max) => {

		value = value.trim();


		if (value.endsWith('%')) {

			return Math.min(Number.parseFloat(value) * max / 100, max);

		}


		return Math.min(Number.parseFloat(value), max);

	};


	const red = parseValue(parts[0], 255);

	const green = parseValue(parts[1], 255);

	const blue = parseValue(parts[2], 255);

	let alpha;


	if (parts.length === 4) {

		alpha = parseValue(parts[3], 1);

	}


	return [red, green, blue, alpha];

};
```

See the code in that repo?

Reply:
Yea I saw, what do you think bout adding that dep or copy paste codes?

Reply:
Dep

Reply:
Yea, Resolved on https://github.com/baseplate-admin/coreproject-v4-ui/pull/64/commits/d4fac7e9f388c23abc80edfe1cdbd9270a5047de